File: apps/api/app/schemas/broadcast.py
# why: 모듈 역할과 책임을 명확히 하기 위한 진입 주석
from datetime import datetime
from pydantic import BaseModel
from app.models.broadcast_slot import BroadcastStatus
from app.schemas.common import TimestampSchema


class BroadcastOut(TimestampSchema):
    """방송 슬롯 응답 스키마."""

    id: int
    channel_id: int
    channel_code: str | None = None
    channel_name: str | None = None
    source_code: str
    start_at: datetime
    end_at: datetime
    raw_title: str
    normalized_title: str
    category: str | None = None
    product_url: str | None = None
    live_url: str | None = None
    sale_price: int | None = None
    original_price: int | None = None
    discount_rate: float | None = None
    price_text: str | None = None
    image_url: str | None = None
    status: BroadcastStatus
    slot_hash: str

    class Config:
        from_attributes = True


class BroadcastDetailOut(BroadcastOut):
    """상세 응답 스키마. MVP에서는 BroadcastOut과 동일하지만 확장을 고려해 분리."""

    pass


class PriceHistoryOut(BaseModel):
    """가격 변동 이력 응답."""

    collected_at: datetime
    sale_price: int | None = None
    original_price: int | None = None
    discount_rate: float | None = None

    class Config:
        from_attributes = True

File: apps/web/src/types/broadcast.ts
// why: 파일 책임을 명확히 하고 유지보수를 쉽게 하기 위한 설명 주석
export type BroadcastStatus = "SCHEDULED" | "LIVE" | "ENDED";

export interface Broadcast {
  id: number;
  channel_id: number;
  channel_code?: string | null;
  channel_name?: string | null;
  source_code: string;
  start_at: string;
  end_at: string;
  raw_title: string;
  normalized_title: string;
  category?: string | null;
  product_url?: string | null;
  live_url?: string | null;
  sale_price?: number | null;
  original_price?: number | null;
  discount_rate?: number | null;
  price_text?: string | null;
  image_url?: string | null;
  status: BroadcastStatus;
  slot_hash: string;
}
